### DISCUSSION TITLE
Handle scenario of bots and bad browsers not following datatypes

### DISCUSSION THREAD

**Discussion by rtconner** [livewire/livewire#4738] [1. Help]
I like to set types on my Component properties. And I use a variety of datatypes..

```
class Foobar extends Component
{
public string $message = '';
public int $page = 1;
public array $category = [];
```

And my UI works and everything works correct for all testing I do.

However, it turns out there are some crazy bots and browsers out there that insist on hitting my website with invalid data types. (Probably trying to hack me or whatever). 

`Cannot assign bool to property App\Http\Livewire\Foobar::$category of type array`

So my error logs fill up, and I get lots of false alarms. It's not a good practice to learn to ignore errors in your logs.

So ... line 50 of HandlesActions..
`$this->{$name} = $value;`

maybe should be updated to like ..

```
# (crappy psuedo code) 
if(gettype($this->{$name}) == 'bool') {
  $this->{$name} = boolvalue($value);
} elseif(gettype($this->{$name}) == 'array') {
  $this->{$name} = arrayvalue($value);
} else {
  $this->{$name} = $value;
}
```

**Comment by joshhanley** [comment]
@rtconner yeah there is currently a PR #4154 being worked on which will enable better support for things like that.

In the short term you can actually make use of `updating` hooks and passing the value by reference to do this yourself in your components to update the value before it is assigned to the property.

For example (notice value is passed by reference `&`)

```php
public function updatingPage(&$value)
{
    $value = (int) $value;
}
```

Hope this helps!
